tests/test_utils.rs

```diff
@@ -26,31 +21,79 @@ use alloc::{
 };
 use core::fmt::Debug;
 
-use crate::ast::*;
-use crate::dialect::*;
-use crate::parser::{Parser, ParserError};
-use crate::tokenizer::Tokenizer;
+use sqlparser::dialect::*;
+use sqlparser::parser::{Parser, ParserError};
+use sqlparser::{ast::*, tokenizer::Token};
+
+#[macro_export]
+macro_rules! nest {
+    ($base:expr $(, $join:expr)*) => {
+        TableFactor::NestedJoin(Box::new(TableWithJoins {
+            relation: $base,
+            joins: vec![$(join($join)),*]
+        }))
+    };
+}
+
+pub trait Parse {
```

Comment:
I think you are using this trait so you can have a `dyn Parse` and dispatch dynamically at runtime what dialect is being used. This is likely something other users of the crate might want / might be using (dynamically pick a dialect).

What would you think about putting this trait in the main  crate (or perhaps even calling it `Parser` and changing the struct to `ParserImpl`)?

Reply:
I highly doubt that any API users actually need `Parser` trait objects. Note that with generics it's still possible to "dynamically" choose which dialect is being used at runtime, you don't need trait objects for that. As an example for that see the updated `examples/cli.rs`.

Reply:
> As an example for that see the updated examples/cli.rs.

Indeed -- I think in the case of `examples/cli.rs` the distinction is that the call to `parse<D>` is now also generic (the generics perolate back up)